src/serac/infrastructure/debug_print.hpp

```diff
@@ -68,53 +68,48 @@ std::ostream& operator<<(std::ostream& out, serac::SignedIndex i)
 }
 
 /**
- * @brief write a 2D array of values out to file, in a space-separated format
+ * @brief write an array of values out to file, in a space-separated format
  * @tparam T the type of each value in the array
  * @param v the values to write to file
  * @param filename the name of the output file
  */
-template <typename T>
-void write_to_file(axom::Array<T, 2, axom::MemorySpace::Host> arr, std::string filename)
+template <typename ArrayT>
+void write_to_file(const ArrayT& arr_orig, std::string filename)
 {
   std::ofstream outfile(filename);
 
-  for (axom::IndexType i = 0; i < arr.shape()[0]; i++) {
-    outfile << "{";
-    for (axom::IndexType j = 0; j < arr.shape()[1]; j++) {
-      outfile << arr(i, j);
-      if (j < arr.shape()[1] - 1) outfile << ", ";
-    }
-    outfile << "}\n";
-  }
-
-  outfile.close();
-}
-
-/**
- * @brief write a 3D array of values out to file, in a mathematica-compatible format
- * @tparam T the type of each value in the array
- * @param v the values to write to file
- * @param filename the name of the output file
- */
-template <typename T>
-void write_to_file(axom::Array<T, 3, axom::MemorySpace::Host> arr, std::string filename)
-{
-  std::ofstream outfile(filename);
+#ifdef SERAC_USE_UMPIRE
+  axom::Array<ArrayT::value_type, ArrayT::Dims, axom::MemorySpace::Host> arr(arr_orig);
+#else
+  const auto& arr = arr_orig;
+#endif
 
-  outfile << std::setprecision(16);
+  if constexpr(ArrayT::Dims == 2)
+  {
+    for (axom::IndexType i = 0; i < arr.shape()[0]; i++) {
+      outfile << "{";
+      for (axom::IndexType j = 0; j < arr.shape()[1]; j++) {
+        outfile << arr(i, j);
+        if (j < arr.shape()[1] - 1) outfile << ", ";
+      }
+      outfile << "}\n";
+    }
+  } else if constexpr(ArrayT::Dims == 3) {
```

Comment:
We got compiler errors when we have `else if constexpr` before. Can you remove the `else` and make it just two `if constexpr` blocks?